src/SAML2/Provider/HostedIdentityProviderProcessor.php

```diff
@@ -446,14 +464,14 @@ public function continueSingleLogoutService()
             $this->stateHandler->apply(SamlStateHandler::TRANSITION_SLS_START_PROPAGATE);
 
             // Dispatch logout to service providers
-            $sp = $this->serviceProviderRepository->getServiceProvider($state->popServiceProviderIds());
+            $sp = $this->serviceProviderRepository->getServiceProvider($state->getRequest()->getIssuer());
```

Comment:
I don't understand how it should work. I will return it. As I see we will only try to logout  only from the last SP.